src/cmap/connection.ts

```diff
@@ -369,7 +372,28 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
 
     // always emit the message, in case we are streaming
     this.emit('message', message);
-    const operationDescription = this[kQueue].get(message.responseTo);
+    let operationDescription = this[kQueue].get(message.responseTo);
+
+    if (!operationDescription && this.isMonitoringConnection) {
+      // This is how we recover when the initial hello's requestId is not
+      // the responseTo when hello responses have been skipped:
+
+      // First check if the map is of invalid size
+      if (this[kQueue].size > 1) {
```

Comment:
Would it make sense to move this check out of the conditional?  It seems like when it's not a monitoring connection the same condition should hold, since we don't multiplex connections

Reply:
If moved the check up now.

Reply:
Ostensibly, the answer here is "yes", but that would be a potential behavior change that might be better reserved for NODE-4820 and at a mininum would require additional tests that feel out of scope for the actual fix

Reply:
Moved back into the monitoring connection block due to change stream resumability being affected by that change.

Reply:
Interesting.  I'd like to understand how how, but can discuss outside the PR.